jonasfj/dependabot-core
# file parser spec

- Resolve properties and directories from included files
- test: add test to reproduce multiple version defect
- test: ensure parser does not raise an error
- fix: loop through each poetry requirement
- test: assert the expected packages are parsed
- refactor: extract method to parse requirement
- fix: use array literal to prevent converting a Hash to an array of tuples
- style: fix linter errors
- fix: exclude dependencies that do not have at least one supported requirement
- handle terraform module versions with a 'v' prefix
- fix typo: moudules->modules
- Apply suggestions from code review to delete repetitive block
- tidy up pattern override based off PR review comments
- add file updater test cases showing the v prefix being dropped
- Describe workaround for cloning with long file names on Windows
- Clarify that Remote Containers Clone Repository commands are not supported
- Python: Upgrade pyenv to 2.0.6
- Python: Raise resolvability error with explanation when update is not possible
- chore: run builds on a regular basis to detect anomolies
- v0.162.0
- handle support files with dir macro in umbrellas
- bin/dry-run.rb exits outside a developer shell
- Update README
- Prefer checking user over touching a file
- Support basic Kotlin apply, add tests
- Escape paths passed to VendorUpdater
- Bump golang from 1.17 to 1.17.1
- build(deps): bump github.com/dependabot/gomodules-extracted
- fix  `labels': Unsupported provider bitbucket
- Fix minor typos in changelog
- v0.162.1
- Support Gradle files with no top level build.gradle file
- Treat tokens after underscore as numeric if possible
- Simplify string processing
- Initial work on pub support
- Integration with docker, and dart pub
- Minor fixes
- Trying to get dry-run working
- Minor fixes
- Move pub/lib to pub/lib/dependabot
- Missing files
- Allow constraint 'any'
- Use Open3.capture3 with multiple args
- Fix scoping
- Sort dependencies
- Caching of report
- Make raw_constraint optional
- Use cahcing and fix minor issues
- minor fixes
- Ignore replaced dependencies
- Minor fixes
- v0.162.2
- Add support for gradlePluginPortal()
- build(deps-dev): bump jest in /npm_and_yarn/helpers
- Handle .tar path dependency
- Fix indentation
- build(deps): bump composer/composer in /composer/helpers/v1
- build(deps): bump @npmcli/arborist in /npm_and_yarn/helpers
- build(deps-dev): bump friendsofphp/php-cs-fixer in /composer/helpers/v2
- Add issue template config with Dependabot contact links
- Fix elm tests
- Fix Poetry unreachable git deps error
- build(deps-dev): bump prettier in /npm_and_yarn/helpers
- Gradle: Prefer method call over instance variable access in file fetcher
- build(deps): bump poetry from 1.1.7 to 1.1.11 in /python/helpers
- build(deps): bump composer/composer in /composer/helpers/v2
- build(deps): bump pip-tools from 6.2.0 to 6.3.0 in /python/helpers
- build(deps): bump wheel from 0.36.2 to 0.37.0 in /python/helpers
- build(deps-dev): bump phpstan/phpstan in /composer/helpers/v1
- build(deps-dev): bump friendsofphp/php-cs-fixer in /composer/helpers/v1
- v0.163.0
- remove trailing whitespaces
- Recommend `git reset` when cloning repository on Windows fails
- build(deps): bump pip-tools from 6.3.0 to 6.3.1 in /python/helpers
- build(deps-dev): bump jest in /npm_and_yarn/helpers
- Make lockfile optional
- build(deps): bump pip from 21.1.3 to 21.2.4 in /python/helpers
- build(deps): bump flake8 from 3.9.2 to 4.0.0 in /python/helpers
- build(deps-dev): bump eslint in /npm_and_yarn/helpers
- build(deps): bump golang.org/x/mod in /go_modules/helpers
- Support mono-repos/local path-dependencies
- Upgrade OTP to latest minor: 23.3.4.5
- Python: Upgrade pyenv to 2.1.0
- Bump Terraform from 1.0.6 to 1.0.8
- build(deps): bump pip-tools from 6.3.1 to 6.4.0 in /python/helpers
- build(deps): bump flake8 from 4.0.0 to 4.0.1 in /python/helpers
- update Elixir from 1.12.2 -> 1.12.3
- update erlang solutions from 1.0 to 2.0
- v0.163.1
- Add license to image and gemspec
- Add license to gem build directory
- Set license back to Nonstandard
- Allow passing target_project_id to Gitlab pr creator and updater
- Revert "Add license to gem build directory"
- Update specs
- Move target_project_id to provider_metadata
- Treat GHES hosted sources as github sources
- Revert "v0.163.1"
- v0.164.0
- Ensure we cleanup tmp directories after use
- Only check auth for github.com
- Always require a full repository clone
- v0.164.1
- feat: specify timeouts per spawned process
- style: fix linter errors
- test: update glass assertions
- test: ensure the proper command is generated
- feat: trap SIGHUB and flush error to stdout
- refactor: collapse multi line conditional
- style: fix linter errors
- refactor: extract class to build shell command
- style: fix linter errors
- fix: ensure min of 1 minute and max of 30 minutes per op
- refactor: make timeout_seconds a private method
- Respect ignored_versions
- refactor: use Comparable#clamp
- v0.165.0
- Bump to go 1.17.3
- Move composer-not-found fixture from decommissioned dependabot.com
- Use new pub binary
- Handle '>=a.b.c <d.e.f' constraints
- Ignore errors from Source enterprise check and ignore known failures
- Explicitly ignore metadata detection for fuchsia.googlesource.com
- v0.166.0
- Improve error handling when `terraform init` fails
- Rescue terraform registry connection errors
- v0.166.1
- nits
- Flutter support (#1)
- Update nomenclature
- Fix condition
- Initial tests
- file_fetcher_spec
- Fix lints
- Revert "Fix lints"
- fix lints (take 2)
- Tests for file_parser
